cirq/testing/lin_alg_utils.py

```diff
@@ -70,8 +70,9 @@ def random_orthogonal(dim: int) -> np.ndarray:
         http://arxiv.org/abs/math-ph/0609050
     """
     m = np.random.randn(dim, dim)
-    q, _ = np.linalg.qr(m)
-    return q
+    q, r = np.linalg.qr(m)
+    d = np.diag(r)
+    return q * (d / abs(d))
```

Comment:
Is d guaranteed not to be zero? We could use `np.sign` instead.

Reply:
I guess the assumption is that the probability of `m` being singular is negligible. IIUC, it's not strictly speaking impossible, but is very, very unlikely. Perhaps we should just assert that `abs(d) > 0`.

BTW: This concern applies in `random_unitary()` above, too.

Reply:
Good enough